diff --git a/automotive/basicmath/Makefile b/automotive/basicmath/Makefile
index 518f84c..9d6cab1 100755
--- a/automotive/basicmath/Makefile
+++ b/automotive/basicmath/Makefile
@@ -1,12 +1,12 @@
 FILE1 = basicmath_small.c   rad2deg.c  cubic.c   isqrt.c
 FILE2 = basicmath_large.c   rad2deg.c  cubic.c   isqrt.c
-
+CC ?= gcc
 all: basicmath_small basicmath_large
 
 basicmath_small: ${FILE1} Makefile
-	gcc -static -O3 ${FILE1} -o basicmath_small -lm
+	$(CC) -static -O3 ${FILE1} -o basicmath_small -lm
 basicmath_large: ${FILE2} Makefile
-	gcc -static -O3 ${FILE2} -o basicmath_large -lm
+	$(CC) -static -O3 ${FILE2} -o basicmath_large -lm
 
 clean:
 	rm -rf basicmath_small basicmath_large output*
diff --git a/automotive/bitcount/Makefile b/automotive/bitcount/Makefile
index 0f42708..736cb7b 100755
--- a/automotive/bitcount/Makefile
+++ b/automotive/bitcount/Makefile
@@ -1,7 +1,7 @@
 FILE = bitcnt_1.c bitcnt_2.c bitcnt_3.c bitcnt_4.c bitcnts.c bitfiles.c bitstrng.c bstr_i.c 
-
+CC?=gcc
 bitcnts: ${FILE} Makefile
-	gcc -static ${FILE} -O3 -o bitcnts
+	$(CC) -static ${FILE} -O3 -o bitcnts
 
 clean:
 	rm -rf bitcnts output*
diff --git a/automotive/qsort/Makefile b/automotive/qsort/Makefile
index 892a75c..6e6e0da 100755
--- a/automotive/qsort/Makefile
+++ b/automotive/qsort/Makefile
@@ -1,12 +1,12 @@
 FILE1 = qsort_small.c
 FILE2 = qsort_large.c
-
+CC?=gcc
 all: qsort_small qsort_large
 
 qsort_small: qsort_small.c Makefile
-	gcc -static qsort_small.c -O3 -o qsort_small -lm
+	$(CC) -static qsort_small.c -O3 -o qsort_small -lm
 qsort_large: qsort_large.c Makefile
-	gcc -static qsort_large.c -O3 -o qsort_large -lm
+	$(CC) -static qsort_large.c -O3 -o qsort_large -lm
 
 clean:
 	rm -rf qsort_small qsort_large output*
diff --git a/automotive/susan/Makefile b/automotive/susan/Makefile
index 235a7ad..239aea5 100755
--- a/automotive/susan/Makefile
+++ b/automotive/susan/Makefile
@@ -1,5 +1,6 @@
+CC?=gcc
 susan: susan.c Makefile
-	gcc -static -O4 -o susan susan.c -lm 
+	$(CC) -static -O4 -o susan susan.c -lm 
 
 clean:
 	rm -rf susan output*
diff --git a/consumer/jpeg/jpeg-6a/Makefile b/consumer/jpeg/jpeg-6a/Makefile
index ce0c45f..acc5972 100755
--- a/consumer/jpeg/jpeg-6a/Makefile
+++ b/consumer/jpeg/jpeg-6a/Makefile
@@ -20,10 +20,10 @@ manext = 1
 mandir = $(prefix)/man/man$(manext)
 
 # The name of your C compiler:
-CC= gcc
+CC?= gcc
 
 # You may need to adjust these cc options:
-CFLAGS= -O  -I$(srcdir)
+CFLAGS += -O  -I$(srcdir)
 # Generally, we recommend defining any configuration symbols in jconfig.h,
 # NOT via -D switches here.
 # However, any special defines for ansi2knr.c may be included here:
diff --git a/consumer/lame/lame3.70/Makefile b/consumer/lame/lame3.70/Makefile
index 02f06d7..d84db8e 100755
--- a/consumer/lame/lame3.70/Makefile
+++ b/consumer/lame/lame3.70/Makefile
@@ -11,7 +11,7 @@ ARCH = $(shell uname -m)
 
 # defaults:
 PGM = lame
-CC = gcc
+CC ?= gcc
 CC_OPTS =  -O
 GTK = 
 GTKLIBS = 
diff --git a/network/dijkstra/Makefile b/network/dijkstra/Makefile
index 9ebf160..50ca7fc 100755
--- a/network/dijkstra/Makefile
+++ b/network/dijkstra/Makefile
@@ -1,10 +1,11 @@
+CC?=gcc
 all:  dijkstra_large dijkstra_small
 
 dijkstra_large: dijkstra_large.c Makefile
-	gcc -static dijkstra_large.c -O3 -o dijkstra_large
+	$(CC) -static dijkstra_large.c -O3 -o dijkstra_large
 
 dijkstra_small: dijkstra_small.c Makefile
-	gcc -static dijkstra_small.c -O3 -o dijkstra_small 
+	$(CC) -static dijkstra_small.c -O3 -o dijkstra_small 
 
 clean:
 	rm -rf dijkstra_large dijkstra_small output*
diff --git a/network/patricia/Makefile b/network/patricia/Makefile
index acf23a1..b4984c9 100755
--- a/network/patricia/Makefile
+++ b/network/patricia/Makefile
@@ -1,5 +1,6 @@
+CC?=gcc
 patricia: patricia.c patricia_test.c
-	gcc -static patricia.c patricia_test.c -O3 -o patricia
+	$(CC) -static patricia.c patricia_test.c -O3 -o patricia
 
 clean:
 	rm -rf patricia output*
diff --git a/office/stringsearch/Makefile b/office/stringsearch/Makefile
index df0d09a..5ccda4f 100755
--- a/office/stringsearch/Makefile
+++ b/office/stringsearch/Makefile
@@ -1,13 +1,13 @@
 FILE1 = bmhasrch.c  bmhisrch.c  bmhsrch.c  pbmsrch_small.c 
 FILE2 = bmhasrch.c  bmhisrch.c  bmhsrch.c  pbmsrch_large.c 
-
+CC?=gcc
 all: search_small search_large
 
 search_small: ${FILE1} Makefile
-	gcc -static ${FILE1} -O3 -o search_small
+	$(CC) -static ${FILE1} -O3 -o search_small
 
 search_large: ${FILE2} Makefile
-	gcc -static ${FILE2} -O3 -o search_large
+	$(CC) -static ${FILE2} -O3 -o search_large
 
 clean:
 	rm -rf search_small search_large output*
diff --git a/security/blowfish/Makefile b/security/blowfish/Makefile
index 334bb20..6aebcca 100755
--- a/security/blowfish/Makefile
+++ b/security/blowfish/Makefile
@@ -20,8 +20,8 @@ MAKE=make -f Makefile
 #CC=cc
 #CFLAG= -O
 
-CC=gcc
-CFLAG= -static -O3 -fomit-frame-pointer
+CC?=gcc
+CFLAG += -static -O3 -fomit-frame-pointer
 
 CFLAGS=$(OPTS) $(CFLAG)
 CPP=$(CC) -E
diff --git a/telecomm/CRC32/Makefile b/telecomm/CRC32/Makefile
index f5862e9..6018311 100755
--- a/telecomm/CRC32/Makefile
+++ b/telecomm/CRC32/Makefile
@@ -1,5 +1,6 @@
+CC?=gcc
 crc: crc_32.c Makefile
-	gcc -static crc_32.c -O3 -o crc 
+	$(CC) -static crc_32.c -O3 -o crc 
 
 clean:
 	rm -rf crc output*
diff --git a/telecomm/FFT/Makefile b/telecomm/FFT/Makefile
index 7645943..c8e4953 100755
--- a/telecomm/FFT/Makefile
+++ b/telecomm/FFT/Makefile
@@ -1,15 +1,15 @@
 OBJ = main.o fftmisc.o  fourierf.o
 FILE = main.c fftmisc.c  fourierf.c
 CFLAGS = -static -O3 -g
-
+CC?=gcc
 fft: ${OBJ} Makefile
-	gcc  ${CFLAGS} ${OBJ} -o fft -lm
+	$(CC)  ${CFLAGS} ${OBJ} -o fft -lm
 fftmisc.o: fftmisc.c
-	gcc ${CFLAGS} -c fftmisc.c
+	$(CC) ${CFLAGS} -c fftmisc.c
 fourierf.o: fourierf.c
-	gcc ${CFLAGS} -c fourierf.c
+	$(CC) ${CFLAGS} -c fourierf.c
 main.o: main.c
-	gcc ${CFLAGS} -c main.c
+	$(CC) ${CFLAGS} -c main.c
 
 clean:
 	rm -rf *.o fft output*
diff --git a/telecomm/adpcm/src/Makefile b/telecomm/adpcm/src/Makefile
index a2b04f1..061b716 100755
--- a/telecomm/adpcm/src/Makefile
+++ b/telecomm/adpcm/src/Makefile
@@ -11,7 +11,7 @@
 # Also, on Sun machines, in order to use the SPIXTOOLS and SHADE,
 # uncomment this option: -static.
 #
-CC = gcc
+CC = gcc
 LFLAGS = -mv8
 LFLAGS = $(LFLAGS) -static
 #
