diff --git a/office/ispell/correct.c b/office/ispell/correct.c
index ff5f9df..f82bcbb 100755
--- a/office/ispell/correct.c
+++ b/office/ispell/correct.c
@@ -161,7 +161,7 @@ static void	save_root_cap P ((ichar_t * word, ichar_t * pattern,
 		  struct flagent * sufent,
 		  ichar_t savearea[MAX_CAPS][INPUTWORDLEN + MAXAFFIXLEN],
 		  int * nsaved));
-static char *	getline P ((char * buf));
+static char *	lgetline P ((char * buf));
 void		askmode P ((void));
 void		copyout P ((char ** cc, int cnt));
 static void	lookharder P ((char * string));
@@ -443,7 +443,7 @@ checkagain:
 
 		move (li - 1, 0);
 		(void) putchar ('!');
-		if (getline (buf) == NULL)
+		if (lgetline (buf) == NULL)
 		    {
 		    (void) putchar (7);
 		    erase ();
@@ -468,7 +468,7 @@ checkagain:
 		    (void) printf ("%s ", CORR_C_READONLY);
 		    }
 		(void) printf (CORR_C_REPLACE_WITH);
-		if (getline (ctok) == NULL)
+		if (lgetline (ctok) == NULL)
 		    {
 		    (void) putchar (7);
 		    /* Put it back */
@@ -530,7 +530,7 @@ checkagain:
 		char	buf[100];
 		move (li - 1, 0);
 		(void) printf (CORR_C_LOOKUP_PROMPT);
-		if (getline (buf) == NULL)
+		if (lgetline (buf) == NULL)
 		    {
 		    (void) putchar (7);
 		    erase ();
@@ -1384,7 +1384,7 @@ static void save_root_cap (word, pattern, prestrip, preadd, sufstrip, sufadd,
 #endif /* NO_CAPITALIZATION_SUPPORT */
     }
 
-static char * getline (s)
+static char * lgetline (s)
     register char *	s;
     {
     register char *	p;
