diff --git a/telecomm/adpcm/src/Makefile b/telecomm/adpcm/src/Makefile
index a2b04f1..be7448d 100755
--- a/telecomm/adpcm/src/Makefile
+++ b/telecomm/adpcm/src/Makefile
@@ -26,7 +26,6 @@ ALLSRC=README adpcm.2 Makefile adpcm.h libst.h adpcm.c timing.c rawcaudio.c \
 	rawdaudio.c suncaudio.c sgicaudio.c sgidaudio.c hello.adpcm.uu
 
 all: adpcm.o rawcaudio rawdaudio timing
-	cp rawcaudio rawdaudio timing ../bin/.
 
 sgi: all sgicaudio sgidaudio
 
