From 2a6292acd165f33a8d08745468b72368a6d36d6c Mon Sep 17 00:00:00 2001
From: max <guoqun.ma@spacemit.com>
Date: Mon, 30 Oct 2023 20:28:26 +0800
Subject: [PATCH] update Make arch file

---
 unix/Make-arch | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/unix/Make-arch b/unix/Make-arch
index 1044370..700154f 100644
--- a/unix/Make-arch
+++ b/unix/Make-arch
@@ -1187,6 +1187,19 @@ linux-debug:
 	"RANLIB = ranlib" \
 	"LIBS = -L. -ltachyon $(MISCLIB) -lm"
 
+# Linux on AMD64/EM64T, using gcc
+linux-riscv64:
+	$(MAKE) all \
+	"ARCH = linux-riscv64" \
+	"CC = $(CC)" \
+	"CFLAGS = -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux  $(MISCFLAGS)" \
+	"AR = ar" \
+	"ARFLAGS = r" \
+	"STRIP = touch" \
+	"RANLIB = ranlib" \
+	"LIBS = -L. -ltachyon $(MISCLIB) -lm"
+
+
 # Linux on AMD64/EM64T, using gcc
 linux-64:
 	$(MAKE) all \
-- 
2.25.1

