diff --git a/Makefile b/Makefile
index 1de00e7..3405f5a 100644
--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,6 @@ DEFINES=-DUSE_LARGEFILES
 # in process scope despite of threads
 # DEFINES=-DGETRUSAGE_PROCESS_SCOPE
 
-LINK=gcc
 TIOTEST=tiotest
 TEST_LARGE=test_largefiles
 PROJECT=tiobench
@@ -38,13 +37,13 @@ test_largefiles.o: tiotest.h test_largefiles.c
 	$(CC) -c -DUSE_LARGEFILES $(CFLAGS) $(DEFINES) test_largefiles.c -o test_largefiles.o
 
 $(TIOTEST): tiotest.o crc32.o
-	$(LINK) -o $(TIOTEST) $(LDFLAGS) tiotest.o crc32.o -lpthread
+	$(CC) -o $(TIOTEST) $(LDFLAGS) tiotest.o crc32.o -lpthread
 	@echo
 	@echo "./tiobench.pl --help for usage options"
 	@echo
 
 $(TEST_LARGE): test_largefiles.o
-	$(LINK) -o $(TEST_LARGE) test_largefiles.o
+	$(CC) -o $(TEST_LARGE) test_largefiles.o
 
 clean:
 	rm -f test_largefiles.o tiotest.o crc32.o $(TIOTEST) $(TEST_LARGE) core
