From 92b9664deffcf98046373700af34d3893bd3e097 Mon Sep 17 00:00:00 2001
From: max <guoqun.ma@spacemit.com>
Date: Mon, 30 Oct 2023 17:39:46 +0800
Subject: [PATCH] do not build tornado

---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 2271fdc..5b4791b 100644
--- a/Makefile
+++ b/Makefile
@@ -19,6 +19,8 @@ endif
 
 CC?=gcc
 
+DONT_BUILD_TORNADO=1
+
 COMPILER = $(shell $(CC) -v 2>&1 | grep -q "clang version" && echo clang || echo gcc)
 GCC_VERSION = $(shell echo | $(CC) -dM -E - | grep __VERSION__  | sed -e 's:\#define __VERSION__ "\([0-9.]*\).*:\1:' -e 's:\.\([0-9][0-9]\):\1:g' -e 's:\.\([0-9]\):0\1:g')
 CLANG_VERSION = $(shell $(CC) -v 2>&1 | grep "clang version" | sed -e 's:.*version \([0-9.]*\).*:\1:' -e 's:\.\([0-9][0-9]\):\1:g' -e 's:\.\([0-9]\):0\1:g')
-- 
2.25.1

